ui/lib/UpdateStream.tsx

```diff
@@ -1,7 +1,62 @@
-import * as React from "react";
+import React, { useState } from "react";
+import StreamManager from "../../sdk/js/index";
+import { Address } from "viem";
 
-const UpdateStream = () => {
-  return <div />;
+interface UpdateStreamProps {
+  creator: Address;
+  streamId: number;
+  sm: StreamManager;
+}
+
+const UpdateStream: React.FC<UpdateStreamProps> = (props) => {
+  // Set the amount you want to add
+  const [amount, setAmount] = useState<number>(0);
+  // Get results of the transaction to display to the user
+  const [result, setResult] = useState<string | null>(null);
+
+  const handleUpdate = async () => {
+    try {
+      const result = await props.sm.update(
+        props.creator,
+        props.streamId,
+        amount
+      );
+      setResult(`Stream updated. New Time left: ${result}`);
+    } catch (error) {
+      if (error instanceof Error) {
+        setResult(`Error updating stream: ${error.message}`);
+      } else {
+        setResult(`Error updating stream: ${error}`);
+      }
+    }
+  };
+
+  return (
+    <div className="update-stream-container">
+      <div className="update-stream-title">
+        {" "}
+        Enter the amount you want to add to the stream:{" "}
+      </div>
+      <input
+        type="number"
+        value={amount}
+        onChange={(e) => setAmount(Number(e.target.value))}
```

Comment:
issue: `amount` here is likely something a human is going to type (e.g. "1.5" to mean 1.5 USDC), but the contract isn't going to read them that way because of the false-decimal handling.  We should be handling human-to-real conversions here, so if someone inputs 22.50 USDC, we would converting it to `22500000` as a BigInt for a 6-decimal token (decimals should be read from the contract or set in config, depending) before sending it to the contract.  IMO, we should store everything as bigints and only render for humans.

Veim's [`parseEther()`](https://viem.sh/docs/utilities/parseEther.html) and [`praseUnits()`](https://viem.sh/docs/utilities/parseUnits.html) and their equivalent format* functions are probably your friends here.

Let me know if you'd like more context on this.